src/java/main/org/apache/zookeeper/server/quorum/QuorumPeerMain.java

```diff
@@ -141,6 +145,16 @@ public void runFromConfig(QuorumPeerConfig config)
 
       LOG.info("Starting quorum peer");
       try {
+          MetricsProvider metricsProvider;
+          try {
+              metricsProvider = MetricsProviderBootstrap
+                      .startMetricsProvider(config.metricsProviderClassName, new Properties());
+          } catch (MetricsProviderLifeCycleException error) {
+              LOG.error("Cannot boot MetricsProvider {}", config.metricsProviderClassName, error);
+              throw new IOException("Cannot boot MetricsProvider " + config.metricsProviderClassName,
```

Comment:
Services like Prometheus Client will open an http endpoint (in our specific case I think we will leverage existing admin server, but anyway the registration in the servlet container may fail), or some other service will try to bind to JMX, so it is possible that something goes wrong during start.
You cannot start a server without metrics, otherwise monitoring system won't have control of what is going on.